core/src/message/stored.rs

```diff
@@ -216,3 +216,59 @@ impl Deref for StoredDispatch {
         self.message()
     }
 }
+
+impl From<StoredDelayedDispatch> for StoredDispatch {
+    fn from(dispatch: StoredDelayedDispatch) -> Self {
+        StoredDispatch::new(dispatch.kind, dispatch.message, None)
+    }
+}
+
+/// Stored message with entry point.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+pub struct StoredDelayedDispatch {
```

Comment:
Having a separate type for stored delayed dispatch is a good practice, but introducing it only for the migration seems odd.

Except for tests, we can check the `None` with `post_upgrade`, isn't it?
Instead of introducing a temporary type, I would recommend even make fields of `StoredDispatch` public until the migration is applied to the real chain and then revert this (also with migrations deletion).

@breathx what do you think?

Reply:
@techraed,

The `StoredDelayedDispatch` is not temporary solution and even more is not for this migrations.

Dispatches that are just sent with delay are never have `ContextStore`. It's much better to see type-level guarantees for storing data instead of just every time check that it's none.

Implementing it within the PR is preferable way because we can fit two different independent (!) changes on storage in one migration cycle.

Speaking of migrations itself: we will perform transformation with checks: if some of the message context is not None, which is impossible in case of our runtime has no significant bugs, we will log::error it

Reply:
> @techraed,
> 
> The `StoredDelayedDispatch` is not temporary solution and even more is not for this migrations.
> 
> Dispatches that are just sent with delay are never have `ContextStore`. It's much better to see type-level guarantees for storing data instead of just every time check that it's none.
> 
> Implementing it within the PR is preferable way because we can fit two different independent (!) changes on storage in one migration cycle.
> 
> Speaking of migrations itself: we will perform transformation with checks: if some of the message context is not None, which is impossible in case of our runtime has no significant bugs, we will log::error it

discussed, thank you.

Reply:
This definitely should be described in comments